client/src/js/services/VisitsService.js

```diff
@@ -0,0 +1,29 @@
+angular.module('bhima.services')
+.service('VisitService', VisitService);
+
+VisitService.$inject = [ '$http', 'util' ];
+
+function VisitService($http, util) {
+  var service = this;
+  var baseUrl = '/patients/';
+ 
+  service.read = read;
+  service.checkin = checkin; 
+  
+  function read(patientUuid, options) { 
```

Comment:
This would be a good thing to standardise. 

This particular return was just to stop the service from reporting the confusing errors with 'visit' or 'checkin' not being valid patient UUIDs, however this will in turn just give the error 'undefined has no method `then`' or something like that. 

We could standardise this to either allow the $http response to send back a potentially confuisng error or to throw a client side error. 

For now I don't see an issue with getting the $http error - the developer should clearly see they have not provided a UUID.